internal/cluster/validator.go

```diff
@@ -426,7 +426,9 @@ func (v *clusterValidator) sufficientMastersCount(c *clusterPreprocessContext) V
 	}
 
 	//validate worker candidates count
-	if len(workers) == common.IllegalWorkerHostsCount {
+	// Allow for 1 worker to be added if masters are schedulable. Otherwise,
+	// validation should fail.
+	if !common.AreMastersSchedulable(c.cluster) && len(workers) == common.IllegalWorkerHostsCount {
```

Comment:
no tests?

Reply:
I marked the PR as WIP, wrote this in the description:

![image](https://user-images.githubusercontent.com/13816/151147953-9838d759-1be7-4ca5-a35e-ae90c5329986.png)

And then held it:

![image](https://user-images.githubusercontent.com/13816/151147989-a4b4f988-670d-4116-8a48-c71c2fc3b9af.png)

This is all to say that yeah, I will add tests and the PR is not ready :rofl: :hugs:

Reply:
missed that part, thanks